### PR TITLE
Add download support for tar.gz & don't download data if exists 

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fix: Don't download data if exists. Will throw an error if Data exists in the working directory, and there's no internet.

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/core/data/utils.py [hunk 1]
```diff
@@ -34,16 +34,16 @@ def download_file(url: str, path: str, verbose: bool = False) -> None:
...
 
     if not os.path.exists(local_filename):
+        r = requests.get(url, stream=True)
+        file_size = int(r.headers['Content-Length']) if 'Content-Length' in r.headers else 0
+        chunk = 1
+        chunk_size = 1024
+        num_bars = int(file_size / chunk_size)
+        if verbose:
+            print(dict(file_size=file_size))
+            print(dict(num_bars=num_bars))
```
Comment (Borda) [inline flash/core/data/utils.py:46]:
```suggestion
            looging.info(f'file size: {dict(file_size=file_size)} \n # bars: {dict(num_bars=num_bars)}')
```

**Review by Borda — APPROVED** [review]

**Review by teddykoker — APPROVED** [review]

File: flash/core/data/utils.py [hunk 2]
```diff
@@ -34,16 +34,15 @@ def download_file(url: str, path: str, verbose: bool = False) -> None:
...
-        print(dict(num_bars=num_bars))
 
     if not os.path.exists(local_filename):
+        r = requests.get(url, stream=True)
+        file_size = int(r.headers['Content-Length']) if 'Content-Length' in r.headers else 0
+        chunk = 1
+        chunk_size = 1024
+        num_bars = int(file_size / chunk_size)
+        if verbose:
+            logging.info(f'file size: {dict(file_size=file_size)} \n # bars: {dict(num_bars=num_bars)}')
```
Comment (carmocca) [inline flash/core/data/utils.py:45]:
```suggestion
            logging.info(f'file size: {file_size}\n# bars: {num_bars}')
```

File: flash/core/data/utils.py [hunk 3]
```diff
@@ -34,16 +34,15 @@ def download_file(url: str, path: str, verbose: bool = False) -> None:
...
-    chunk = 1
-    chunk_size = 1024
-    num_bars = int(file_size / chunk_size)
-    if verbose:
-        print(dict(file_size=file_size))
-        print(dict(num_bars=num_bars))
 
     if not os.path.exists(local_filename):
+        r = requests.get(url, stream=True)
+        file_size = int(r.headers['Content-Length']) if 'Content-Length' in r.headers else 0
```
Comment (carmocca) [inline flash/core/data/utils.py:40]:
```suggestion
        file_size = int(r.headers.get('Content-Length', 0))
```

Comment (aribornstein) [reply flash/core/data/utils.py:40]: This should be a configurable option I can think of use cases where I might want to redownload if the original file changes.

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by justusschock — APPROVED** [review]

**Comment by aribornstein** [comment]
What's blocking this from being merged?

**Pull request merged**